src/mon/OSDMonitor.cc

```diff
@@ -3632,8 +3632,6 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
       ss << osdmap.get_flag_string(osdmap.get_flags() & warn_flags)
 	 << " flag(s) set";
       summary.push_back(make_pair(HEALTH_WARN, ss.str()));
-      if (detail)
-	detail->push_back(make_pair(HEALTH_WARN, ss.str()));
```

Comment:
i think in general we should keep these.. so that the detail summary is an exhaustive list and you don't need to try to figure out the unique set of things from the union of the summary and detail lists